File: day6/moores.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
           #moores voting algorithm
            
            
        idx =0
        count =1
        for i in range(1, len(nums)):
            if(nums[i] == nums[idx]):       #idx is candidate
                count+=1
            else:
                count-=1
            if(count == 0):              #reset candidate
                idx =i                  # we reset new index to be the current index
                count =1              #so that ir resets to original count 
          
        return nums[idx]                 # the candidate is returned

File: day6/moores.java
package day 6;

// public class sol {
    
// }


class Solution {
    public int majorityElement(int[] nums) {
           //moores voting algorithm
        int idx =0;
        int count =1;
        for(int i =1;i<nums.length; ++i){
            if(nums[i] == nums[idx])       //idx is candidate
                count++;
            else
                count--;
            if(count == 0){              //reset candidate
                idx =i;                  // we reset new index to be the current index
                count =1;               // so that ir resets to original count 
            }
        }
        
      return nums[idx];                  // the candidate is returned
        
    }
}